### PR TITLE
[GH-1024] Make start workload endpoint take a single UUID

### PR DESCRIPTION
**Pull request by rexwangcc**
### Purpose

- https://broadinstitute.atlassian.net/browse/GH-1024
- Discussions:
    - https://github.com/broadinstitute/wfl/pull/164#discussion_r501691419
    - https://broadinstitute.slack.com/archives/CFA6Q2QBU/p1603916497140900

### Changes

- [x] The `/api/v1/start` endpoint now takes a single UUID string or a map that has `:uuid` has its key.
- [x] Specs are updated accordingly.
- [x] Tests are updated accordingly. 
- [x] Docs are updated accordingly.

### Review Instructions

Now you could run something like:

```bash
curl -X "POST" "http://localhost:3000/api/v1/start" \
     -H 'Authorization: Bearer $(gcloud auth print-access-token)' \
     -H 'Content-Type: application/json' \
     -d $'{
                 "uuid": "d0ab1865-556e-4a7e-8f5c-de80d1081529",
             }'
```
to start a single workload, and the response will also be a single map that represents a workload entity. The easiest way to test is to call `create` and `start` sequentially:

```bash
curl -X "POST" "http://localhost:3000/api/v1/create" \
     -H 'Authorization: Bearer $(gcloud auth print-access-token)' \
     -H 'Content-Type: application/json' \
     -d $'{
  "cromwell": "https://cromwell-gotc-auth.gotc-dev.broadinstitute.org",
  "project": "(Test) Exhaustive WFL Code Bashing Test",
  "input": "fake-input",
  "items": [
    {
      "inputs": {
        "dst": "gs://broad-gotc-dev-zero-test/wfl-test-b17d834c-7c27-4e7a-a6df-3e2a4ef4def7/output.txt",
        "src": "gs://broad-gotc-dev-zero-test/wfl-test-b17d834c-7c27-4e7a-a6df-3e2a4ef4def7/input.txt"
      }
    }
  ],
  "output": "fake-output",
  "pipeline": "copyfile"
}'
```
and then use the JSON response body from this request as the payload of the above `/api/v1/start` endpoint.

### Testing

I'm able to run `unit` and `integration` tests fine. When it comes to system tests, While other modules work as expected, the `wgs` module failed due to an empty payload:
```json
{
      "uuid":""
},
```
with errors like: `01:08:38 WARN  wfl.api.spec - Swallowed exception and returned nil in wfl.util/do-or-nil`. I can see where it came from, not sure if I should fix it in this PR or wait for the work @ehigham mentioned he planned/already doing.

**NVM, there's a place in the testing mock `start` handler that I didn't notice, the issues should be fixed by the latest commit. Will merge once the system test pass.**

The system test is passing now, merging:

```
--- system (clojure.test) ---------------------------
wfl.system.v1-endpoint-test
  test-start-workload
    calling api/v1/start with ExternalWholeGenomeReprocessing workload
    calling api/v1/start with AllOfUsArrays workload
    calling api/v1/start with ExternalExomeReprocessing workload
    calling api/v1/start with copyfile workload

1 tests, 28 assertions, 0 failures.
```

### CODE REVIEW

File: api/src/wfl/api/handlers.clj [hunk 1]
```diff
@@ -81,7 +81,8 @@
             (workloads/load-workloads tx)))))))
 
 (defn post-start
-  "Start the workloads with UUIDs in REQUEST."
+  "Start the workload with UUID in REQUEST.
+   Can take either a single UUID or a map with UUID as key."
```
Comment (ehigham) [inline api/src/wfl/api/handlers.clj:85]: Just take one or the other - don't take both!

File: api/src/wfl/api/handlers.clj [hunk 2]
```diff
@@ -91,9 +92,10 @@
                   (logr/infof "starting workload %s" uuid)
                   (workloads/start-workload! tx workload)))))]
     (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
-      (let [uuids (map :uuid (distinct (get-in request [:parameters :body])))]
-        (logr/infof "post-start endpoint called: uuids=%s" uuids)
-        (succeed (mapv (partial go! tx) uuids))))))
+      (let [body  (get-in request [:parameters :body])
+            {uuid :uuid :or {uuid body}} body]
+        (logr/infof "post-start endpoint called: uuid=%s" uuid)
+        (succeed (go! tx uuid))))))
```
Comment (ehigham) [inline api/src/wfl/api/handlers.clj:98]: I'd recommend inlining `go!` for readability - previously it was used to map over uuids, now that's not requrired

File: api/src/wfl/api/spec.clj [hunk 3]
```diff
@@ -108,3 +106,8 @@
 (s/def ::workflow-request (s/keys :req-un [::end
                                            ::environment
                                            ::start]))
+
+;; /api/v1/start
+(s/def ::start-workfload-request (s/or
+                                   :uuid ::uuid
+                                   :uuid-kv ::uuid-kv))
```
Comment (ehigham) [inline api/src/wfl/api/spec.clj:113]: newline

File: api/src/wfl/api/handlers.clj (hunk 2 above)
Comment (rexwangcc) [reply api/src/wfl/api/handlers.clj:98]: will update

File: api/src/wfl/api/handlers.clj (hunk 1 above)
Comment (rexwangcc) [reply api/src/wfl/api/handlers.clj:85]: per our discussion [here](https://broadinstitute.slack.com/archives/CFA6Q2QBU/p1603916497140900), will update this.

**Review by ehigham — APPROVED** [review]
Looks good. Nice one.

**Review by samanehsan — APPROVED** [review]
LGTM! :octocat:

**Pull request merged**